lib/client-base/index.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import type {ChildProcess} from 'child_process'
+import {formatOptions, keybaseExec, keybaseStatus} from '../utils'
+import {API_VERSIONS} from '../constants'
+
+export type Initialized = false | 'paperkey' | 'currentService'
+
+export type InitOptions = {|
+  verbose?: boolean,
+|}
+
+export type ApiCommandArg = {|apiName: string, method: string, options: Object|}
+
+class ClientBase {
+  initialized: Initialized
+  username: void | string
+  devicename: void | string
+  homeDir: string
+  verbose: boolean
+  spawnedProcesses: ChildProcess[]
+
+  constructor() {
+    this.initialized = false
+    this.verbose = false
+    this.spawnedProcesses = []
+  }
+
+  /**
+   * @ignore
+   * Initializes the client to talk to a service.
+   */
+  async _init(homeDir: string, options?: InitOptions): Promise<void> {
+    const initBotInfo = await keybaseStatus(homeDir)
+    this.homeDir = homeDir
+    this.username = initBotInfo.username
+    this.devicename = initBotInfo.devicename
+  }
+
+  /**
+   * Deinitializes the client by killing any spawned processes
+   * @memberof ClientBase
+   */
+  async deinit(): Promise<void> {
```

Comment:
I think we should add an `_` before deinit and hide it in the docs, we don't want users to be able to call `bot.chat.deint()`
```suggestion
   async _deinit(): Promise<void> {
```

Reply:
oh, good call.